tools can only be built with appropriate nlohmann_json version

#163 introduced a CI failure, because it tries to build a target that links to nlohmann_json, which is not explicitly discovered via `find_package` before (or at least not with an appropriate version).

This should be rather easy to fix by putting the `find_package` that is currently in `tests` into a `CMakeLists.txt` further up the directory tree and then dealing with it appropriately.

On that topic... EDM4hep that I build locally works well with nlohmann_json 3.11.2, but it fails when 3.10.2 was used. Are you sure that 3.10 is the minimal requirement? That's the error I get:
```
[ 29%] Building CXX object edm4hep/CMakeFiles/edm4hep.dir/src/ParticleID.cc.o
/builds/azaborow/OpenDataDetector/EDM4hep/edm4hep/src/ParticleID.cc: In function 'void edm4hep::to_json(nlohmann::json&, const edm4hep::ParticleID&)':
/builds/azaborow/OpenDataDetector/EDM4hep/edm4hep/src/ParticleID.cc:138:3: error: no matching function for call to 'nlohmann::basic_json<>::basic_json(<brace-enclosed initializer list>)'
  138 |   };
      |   ^
In file included from /builds/azaborow/OpenDataDetector/EDM4hep/edm4hep/edm4hep/ParticleID.h:15,
                 from /builds/azaborow/OpenDataDetector/EDM4hep/edm4hep/src/ParticleID.cc:4:
/usr/local/include/nlohmann/json.hpp:19687:5: note: candidate: 'nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>::basic_json(nlohmann::basic_json<ObjectType, ArrayType, StringType, BooleanType, NumberIntegerType, NumberUnsignedType, NumberFloatType, AllocatorType, JSONSerializer, BinaryType>&&) [with ObjectType = std::map; ArrayType = std::vector; StringType = std::__cxx11::basic_string<char>; BooleanType = bool; NumberIntegerType = long int; NumberUnsignedType = long unsigned int; NumberFloatType = double; AllocatorType = std::allocator; JSONSerializer = nlohmann::adl_serializer; BinaryType = std::vector<unsigned char>]'
19687 |     basic_json(basic_json&& other) noexcept
      |     ^~~~~~~~~~
/usr/local/include/nlohmann/json.hpp:19687:5: note:   candidate expects 1 argument, 5 provided
[...]
```

Moreover, I have trouble updating  nlohmann_json version because it seems to be always using the version ROOT was compiled with (but maybe that compatibility is needed).

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,7 +102,7 @@ install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE
               ${CMAKE_CURRENT_SOURCE_DIR}/NOTICE
               DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
-find_package(nlohmann_json 3.10)
+find_package(nlohmann_json 3.10.5)
 
 add_subdirectory(edm4hep)
 add_subdirectory(utils)

